tests/main.nf.test

```diff
@@ -0,0 +1,110 @@
+nextflow_pipeline {
+
+    name "Full Pipeline NF-Tests for LegioVUE"
+    script "main.nf"
+
+    //--- Test 1
+    test("Conflicting Input Data Sources Specified") {
+        tag "fail"
+
+        when {
+            params {
+                input = "$baseDir/tests/test_data/input_none.csv"
+                fastq_dir = "$baseDir/tests/test_data/"
+                outdir = "results"
+            }
+        }
+        then {
+            // Status
+            assert workflow.failed
+
+            // Message
+            assert workflow.stdout.contains("ERROR ~ Please provide input data with either: '--input input.csv' or '--fastq_dir <PATH/TO/PAIRED_FASTQS>' but not both")
+        }
+    }
+
+    //--- Test 2
+    test("No Input Data Sources Specified") {
+        tag "fail"
+
+        when {
+            params {
+                outdir = "results"
+            }
+        }
+        then {
+            // Status
+            assert workflow.failed
+
+            // Message
+            assert workflow.stdout.contains("ERROR ~ Please provide input data with either: '--input input.csv' or '--fastq_dir <PATH/TO/PAIRED_FASTQS>'")
+        }
+    }
+
+    //--- Test 3
+    test("Missing Input Fastq Test") {
+        tag "fail"
+
+        when {
+            params {
+                input = "$baseDir/tests/test_data/input.csv"
+                outdir = "results"
+                min_reads = 100
+            }
+        }
+        then {
+            // Status
+            assert workflow.failed
+
+            // Message
+            assert workflow.stdout.contains("ERROR ~ Validation of pipeline parameters failed!")
+        }
+    }
+
+    //--- Test 4
+    test("Stub Run Test") {
```

Comment:
Thanks so much for including this test 😄 

In general, I recommend not using stub runs for testing, but instead creating very small inputs and databases and running the full pipeline including all underlying dependency software.

However, doing that all requires time. This is a good first step and later on a full test can be added in here.

Reply:
Yeah, echoing this. Testing the stub run only tests that the stub part works, not that the pipeline works under normal circumstances.

Reply:
Yeah I thought that it may be a good spot to put it as a test but makes sense not to. I've changed it to actual data